srtcore/core.cpp

```diff
@@ -7936,7 +7938,7 @@ void CUDT::unlose(const CPacket& packet)
         {
             HLOGF(mglog.Debug, "received out-of-band packet seq %d", sequence);
 
-            int seqdiff = abs(CSeqNo::seqcmp(m_iRcvCurrSeqNo, packet.m_iSeqNo));
+            int seqdiff = abs(CSeqNo::seqoff(packet.m_iSeqNo, m_iRcvCurrSeqNo));
```

Comment:
A notable change. What is happening here?

Reply:
Previously it was effectively just `abs(m_iRcvCurrSeqNo, packet.m_iSeqNo)`.

Now it is `m_iRcvCurrSeqNo - packet.m_iSeqNo,` if the difference is within the threshold `m_iSeqNoTH`.

Otherwise
```
if (packet.m_iSeqNo < m_iRcvCurrSeqNo)
	return m_iRcvCurrSeqNo - packet.m_iSeqNo - m_iMaxSeqNo - 1;

return m_iRcvCurrSeqNo - packet.m_iSeqNo + m_iMaxSeqNo + 1;
```

Reply:
As you remember our research session about these functions, `seqoff` and `seqcmp` differ firstly in that the "primary result" is in reverse: `seqcmp(s1, s2)` is like `(s1 - s2)`, `seqoff(s1, s2)` is like `(s2 - s1)`. The other difference is that `seqoff` returns the real distance, while `seqcmp` only a value that has only a correct relationship to 0, but the value itself may be misleading.